Expose monster type on monster info

julealgon opened the pull request:
This PR introduces one extra information when displaying monster details in the bottom info panel: the monster type.

| | Disabled | Enabled |
| --- | --- | --- |
| Normal (on-screen bar) | ![image](https://user-images.githubusercontent.com/461579/111918457-e72fd880-8a63-11eb-9852-ed205a6ba9cb.png) | ![image](https://user-images.githubusercontent.com/461579/111918492-0fb7d280-8a64-11eb-8c4b-59b819afc810.png) |
| Normal (info panel) | ![image](https://user-images.githubusercontent.com/461579/92981945-7edbbb00-f472-11ea-96a0-70bad26de376.png) | ![image](https://user-images.githubusercontent.com/461579/92981165-1808d280-f46f-11ea-8ab8-057af2f7e595.png) | ![image](https://user-images.githubusercontent.com/461579/111918457-e72fd880-8a63-11eb-9852-ed205a6ba9cb.png) |
| Unique (on-screen bar) | ![image](https://user-images.githubusercontent.com/461579/111918472-f9aa1200-8a63-11eb-9683-019e2ce00480.png) | ![image](https://user-images.githubusercontent.com/461579/111918507-23633900-8a64-11eb-8bb1-439eb10865ba.png) |
| Unique | ![image](https://user-images.githubusercontent.com/461579/92981626-09231f80-f471-11ea-9552-eb0718ce4c59.png)  | ![image](https://user-images.githubusercontent.com/461579/92981782-c6ae1280-f471-11ea-8244-077faff8430b.png)|

The information is added to the second info line to preserve space and for line limit considerations.

A new feature toggle was added to control this behavior, `Show monster type information`, disabled by default.

This resolves #805

julealgon commented:
Retry DeepCode

AJenbo commented:
Been disappointed about DeepCode, it's detected very few issues and has errored a lot, it's removed now.

AJenbo commented:
How about having this behind an ini flag?

julealgon commented:
> How about having this behind an ini flag?

We can definitely take a look at that. I'll rebase this later and add the flag.

glebm commented on Source/diablo.cpp:
```diff
@@ -437,6 +437,7 @@ static void SaveOptions()
 	setIniInt("Game", "Auto Equip Shields", sgOptions.Gameplay.bAutoEquipShields);
 	setIniInt("Game", "Auto Equip Jewelry", sgOptions.Gameplay.bAutoEquipJewelry);
 	setIniInt("Game", "Enable All Quests for Single Player", sgOptions.Gameplay.bAllQuests);
+	setIniInt("Game", "Show monster type information", sgOptions.Gameplay.bShowMonsterType);
```
Is there other information other than the type itself?
If not, consider renaming to "Show monster type"

AJenbo replied:
Call it "Show Monster Type" so the casing is also consistent.

julealgon replied:
Renamed. And fixed casing to match other setting names.

julealgon commented:
Rebased and added toggle for new behavior, disabled by default.

I also made the toggle control the same behavior on the on-screen health bar, if it is enabled.

| Toggle Off | Toggle On |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/461579/111918457-e72fd880-8a63-11eb-9852-ed205a6ba9cb.png) | ![image](https://user-images.githubusercontent.com/461579/111918492-0fb7d280-8a64-11eb-8c4b-59b819afc810.png) |
| ![image](https://user-images.githubusercontent.com/461579/111918472-f9aa1200-8a63-11eb-9683-019e2ce00480.png) | ![image](https://user-images.githubusercontent.com/461579/111918507-23633900-8a64-11eb-8bb1-439eb10865ba.png) |
